src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -394,6 +394,30 @@ async def do_delete(self, id):
         await self.middleware.call("datastore.delete", self._config.datastore, id)
         await self._service_change("nfs", "reload")
 
+    @private
+    async def validate_paths(self, data, schema_name, verrors):
+        """
+        Allowing a list of `paths` for NFS shares is for API consistency between
+        Linux and FreeBSD. In Linux case, each path will be treated as a separate
+        export with identical configuration. Since the `locked` property applies
+        to the share as a whole, we enforce same validation on Linux as on FreeBSD
+        (restricting paths to the same filesystem).
```

Comment:
This seems fine for 22.02, but having this restriction only because our locked shares API does not fit with this particular feature of NFS seems excessive. Do we have plans/ticket to fix this in the future?

Reply:
Well, I think right thing to do would be to have only one path per export created in GUI, and do a migration to convert from list to string (creating new shares as needed). Unfortunately, this would have the side-effect of changing the API for sharing.nfs.

Reply:
Do you want me to create a ticket?

Reply:
@themylogin sure, you can do that. Maybe reasonable to put in bluefin.

Reply:
@anodos325 on the other hand - can we just consider the entire share locked if at least one path is locked? This way only users who use dataset encryption will really be affected.